### PR TITLE
[CUB-5083] Enable device name as an updatable property when using update_resources action

### PR DESCRIPTION
**Pull request by murkitsa**
### Manual PR Checklist

- [x] CHANGELOG entry added
- [x] ~Test coverage added~
- [x] Licenses checked
- [x] All related Jira tickets added to PR title
- [x] Description in Jira ticket is clear and up-to-date

[CUB-5083]: https://labforward.atlassian.net/browse/CUB-5083?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/schemata/definitions/workflow-template/action-objects/update-resource.yml [hunk 1]
```diff
@@ -21,6 +22,24 @@ additionalProperties:
...
             - deviceField.id
     - oneOf:
+        - type: object
+          required:
+            - type
+            - attributes
+          properties:
+            type:
+              type: string
+              enum: [device]
```
Comment (kgrhartlage) [inline src/schemata/definitions/workflow-template/action-objects/update-resource.yml:32]: I understand the the current customer case is for devices, but should we allow this for other resources with `name` also? @ignatiusreza

File: src/schemata/definitions/workflow-template/action-objects/update-resource.yml [hunk 2]
```diff
@@ -21,6 +22,24 @@ additionalProperties:
...
+            attributes:
+              type: object
+              additionalProperties: false
+              properties:
+                custom_attributes:
+                  $ref: '#/definitions/script'
+                  description: |
+                    To update a custom attribute, the script must resolve to a hash object with `name`, `type` and `value` attributes. The supported custom attribute data types are `number`, `text`, `boolean` (checkbox), `date`, `datetime`, `file`, `image`, `physicalQuantity`, `relation`, and `duration`.
```
Comment (kgrhartlage) [inline src/schemata/definitions/workflow-template/action-objects/update-resource.yml:40]:
```suggestion
                  description: &custom_attributes_description |
                    To update a custom attribute, the script must resolve to a hash object with `name`, `type` and `value` attributes. The supported custom attribute data types are `number`, `text`, `boolean` (checkbox), `date`, `datetime`, `file`, `image`, `physicalQuantity`, `relation`, and `duration`.
```
let's write this once and use that anchor throughout this file to make sure we are consistent

**Review by kgrhartlage — CHANGES_REQUESTED** [review]

File: src/schemata/definitions/workflow-template/action-objects/update-resource.yml (hunk 2 above)
Comment (kgrhartlage) [reply src/schemata/definitions/workflow-template/action-objects/update-resource.yml:40]: actually we can move this up one level, right?

```yml
custom_attributes: &custom_attributes
  # rest of definition
```

File: src/schemata/definitions/workflow-template/action-objects/update-resource.yml (hunk 1 above)
Comment (kgrhartlage) [reply src/schemata/definitions/workflow-template/action-objects/update-resource.yml:32]: Looking through the list, there doesn't seem to be an obvious need to update most or all other resources in the context of a workflow run. Jared confirmed that he sees no need for that, so let's keep it to devices for now.

File: src/schemata/definitions/workflow-template/action-objects/update-resource.yml [hunk 3]
```diff
@@ -2,7 +2,8 @@
 type: object
 title: Update Resource
 description: |
-  An action to update a resource. It can be used to update the value of custom attributes, and in case of Workflow Runs also update their title.
+  An action to update a resource. It can be used to update the value of custom attributes.
+  In case of Workflow Runs also update their title and name of Device.
```
Comment (kgrhartlage) [inline src/schemata/definitions/workflow-template/action-objects/update-resource.yml:6]:
```suggestion
  An action to update the value of custom attributes of a resource, as well as the title of a Workflow Run, or the name of a device.
```

**Review by kgrhartlage — CHANGES_REQUESTED** [review]

File: src/schemata/definitions/workflow-template/action-objects/update-resource.yml (hunk 2 above)
Comment (murkitsa) [reply src/schemata/definitions/workflow-template/action-objects/update-resource.yml:40]: hmm could you please reference the doc how to use the ampersand as the anchor, I have not seen this syntax before 😳

Comment (kgrhartlage) [reply src/schemata/definitions/workflow-template/action-objects/update-resource.yml:40]: [Anchors and Aliases in the official docs](https://yaml.org/spec/1.2-old/spec.html#id2765878), which seems not very helpful 🥴 . Here's [another google](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/) result with examples that looks better.

The idea is that you mark an anchor with `&anchor_name` and then use that alias to reference it elsewhere with `*anchor_name`. That way we don't have to repeat ourselves so many times in this file.

I'm not 100% sure if we've used this before here and if it plays nice with the parser we use, but I would expect it does. The generated output in the JSON files should look the same as when typing it out everywhere, then it works.

Comment (kgrhartlage) [reply src/schemata/definitions/workflow-template/action-objects/update-resource.yml:40]: like this:

```yml
              properties:
                custom_attributes: &custom_attributes
                  $ref: '#/definitions/script'
                  description: |
                    To update a custom attribute, the script must resolve to a hash object with `name`, `type` and `value` attributes. The supported custom attribute data types are `number`, `text`, `boolean` (checkbox), `date`, `datetime`, `file`, `image`, `physicalQuantity`, `relation`, and `duration`.
                name:
                  $ref: '#/definitions/script'
        - type: object
          required:
            - type
            - attributes
          properties:
            type:
              type: string
              enum: [workflow_run]
            attributes:
              type: object
              additionalProperties: false
              properties:
                custom_attributes: *custom_attributes
                title:
                  $ref: '#/definitions/script'
```

Comment (murkitsa) [reply src/schemata/definitions/workflow-template/action-objects/update-resource.yml:40]: ahh it is yaml specification, not the json schema, thank you!

**Review by kgrhartlage — APPROVED** [review]

**Pull request merged**